runtime/src/accounts_db.rs

```diff
@@ -1609,20 +1612,44 @@ impl AccountsDb {
 
     // Construct a vec of pubkeys for cleaning from:
     //   uncleaned_pubkeys - the delta set of updated pubkeys in rooted slots from the last clean
-    //   zero_lamport_pubkeys - set of all alive pubkeys containing 0-lamport updates
+    //   dirty_stores - set of stores which had accounts removed
     fn construct_candidate_clean_keys(
         &self,
         max_clean_root: Option<Slot>,
         timings: &mut CleanKeyTimings,
     ) -> Vec<Pubkey> {
         let mut zero_lamport_key_clone = Measure::start("zero_lamport_key");
-        let pubkeys = self.accounts_index.zero_lamport_pubkeys().clone();
```

Comment:
Ha interesting, so most of the time in clean was spent cloning out the 11m zero-lamport keys? I would have thought most of the time was spent in the par scan: https://github.com/solana-labs/solana/blob/9388aaca1539a518c52c7c6d9cd9b99fd044ef21/runtime/src/accounts_db.rs#L1672-L1675 and `clean_accounts_older_than_root` call: https://github.com/solana-labs/solana/blob/9388aaca1539a518c52c7c6d9cd9b99fd044ef21/runtime/src/accounts_db.rs#L1751. Good find!

Reply:
No, it's not the cost of the clone itself. This change reduces the number of keys that we look at to something much smaller than 11 million.

The scan and then the dependency checks where I think where each about 50% of the time, around 10-20 seconds each.

The structure of this change is to reduce the number of keys that is input into this function.

Reply:
Ah yup, that makes sense, fewer inputs to the clean.